controller/aks-cluster-config-handler.go

```diff
@@ -184,7 +185,25 @@ func (h *Handler) recordError(onChange func(key string, config *aksv1.AKSCluster
 	return func(key string, config *aksv1.AKSClusterConfig) (*aksv1.AKSClusterConfig, error) {
 		var err error
 		var message string
-		config, err = onChange(key, config)
+
+		err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			// AKS config is being deleted
+			if config == nil {
```

Comment:
We are duplicating this check, look on line#207

Reply:
The reason for this check is to avoid referencing a nil object when we query for a fresh copy. This is possible if the config has been deleted.